packages/api-explorer/lib/generateFakeSchema.js

```diff
@@ -9,27 +9,22 @@ jsf.option({
 })
 
 function generateObjectExample(example, properties) {
-  const generatedExample = { ...properties ? jsf.generate(properties) : {} }
-  // eslint-disable-next-line guard-for-in
+  const generatedExample = properties ? jsf.generate(properties) : {}
   for (const key in example) {
     generatedExample[key] = generateExampleFromType(example[key], example[key])
   }
   return generatedExample
 }
 
 function generateArrayExample(example) {
-  const generatedExample = []
-  example.forEach(item => {
-    generatedExample.push(generateExampleFromType(item, item))
-  })
-  return generatedExample
+  return example.map(item => generateExampleFromType(item, item))
```

Comment:
Why does this take the same argument twice?

Reply:
The listener on the example key, gives me the value of it and the rootSchema it belongs to
If the example key is a primitive type as a string, we return it directly.
If the example key is a structural type we need to check recursive the children.
In this case:
- The first parameter is used to understand the `type`
- The second is only useful if it were an object to get hypothetical `properties`. We repeat twice item word as parameter because we don't know if it's an object.

Reply:
This test help to understand the recursion
```js
{
      title: 'use example outside properties in array of object',
      schema: {
        type: 'object',
        required: ['foo'],
        example: {foo: [{fooChildren: 'nestedBar'}]},
        properties: {
          foo: {
            type: 'array',
            minItems: 1,
            maxItems: 1,
            items: {
              type: 'object',
              properties: {
                fooChildren: {
                  type: 'string',
                }
              }
            }
          },
          randomKey: {
            type: 'string'
          }
        }
      },
      expectation: generatedFakeSchema => {
        expect(generatedFakeSchema).toEqual({foo: [{fooChildren: 'nestedBar'}], randomKey: expect.any(String)})
      }
```

Reply:
I did not properly understand it but ok, it just seems weird to me why we need two arguments